pkg/user/registry/user/etcd/etcd.go

```diff
@@ -63,7 +63,23 @@ func (r *REST) Get(ctx kapi.Context, name string) (runtime.Object, error) {
 			return nil, kerrs.NewForbidden("user", "~", errors.New("Requests to ~ must be authenticated"))
 		}
 		name = user.GetName()
+
+		if ok, _ := validation.ValidateUserName(name, false); !ok {
+			return &api.User{ObjectMeta: kapi.ObjectMeta{Name: name}}, nil
+		}
+
+		obj, err := r.Etcd.Get(ctx, name)
+		if err == nil {
+			return obj, nil
+		}
+
+		if !kerrs.IsNotFound(err) {
+			return nil, err
+		}
+
+		return &api.User{ObjectMeta: kapi.ObjectMeta{Name: name}}, nil
```

Comment:
We agreed that UID should be empty, so that it is possible to distinguish etcd backed user from non-backed user.